src/main/java/com/google/sps/Trip.java

```diff
@@ -25,27 +27,56 @@
  * Trip is the class for storing a single trip (could be multiple days).
  */
 public class Trip {
+
   // Trip cannot be longer than a month (31 days)
   private static final int MAX_NUM_DAYS = 31;
 
+  // Field attributes for the Trip class.
   private String tripName;
+  private String destinationName;
+  private String tripKey;
+  private String imageSrc;
   private LocalDate startDate;
   private LocalDate endDate;
   private int numDays;
-  private List<TripDay> tripDays;
+
+  // Constants to get and put the Entity objects in Datastore.
+  private static final String TRIP = "trip";
+  private static final String TRIP_NAME = "trip_name";
+  private static final String DESTINATION_NAME = "destination_name";
+  private static final String TRIP_KEY = "trip_key";
+  private static final String IMAGE_SRC = "image_src";
+  private static final String START_DATE = "start_date";
+  private static final String END_DATE = "end_date";
 
   /**
    * Creates a new Trip.
    *
    * @param tripName The human-readable name for the trip. Must be non-null.
+   * @param destinationName The name of the destination they are heading.
+   * @param tripKey The key of the trip as created by the Trip Entity.
+   * @param imageSrc The image source / URL to represent the trip.
    * @param startDate The start date for the trip. Must be non-null. Must be in yyyy-MM-dd date format.
    * @param endDate The end date for the trip. Must be non-null. Must be in yyyy-MM-dd date format.
-   * @param tripDays The list of tripDays. Must be non-null.
    */
-  public Trip(String tripName, String startDate, String endDate, List<TripDay> tripDays) {
+  public Trip(String tripName, String destinationName, String tripKey, 
```

Comment:
why are we no longer using TripDay?

Reply:
TripDay isn't a part of the Trip object anymore because when we store the Trip Entity in Datastore, we assign the Trip Entity as the parent of the TripDay Entity. Therefore, it becomes superfluous to also store the TripDay object in the Trip class, as we get the TripDay Entity through Datastore.

This is also because we tried to structure the classes to mimic the Entity objects; the fields in the Trip, TripDay, and Event classes are the properties that are stored for each of their respective Entity objects. This is why the TripDay object also does not store a list of Event objects.

If you think this is a flawed design structure, please let me know! This wouldn't be too difficult to modify.

Reply:
Ah okay, thanks for explaining. I'm fine with it if that's how you've planned the design. It's just a bit hard to visualize without an example!